### PR TITLE
content/docs/next/learning: propose v3.5 auto-promotion in design-learner.md

### PR DESCRIPTION
**Pull request by maxenglander**
(Moved here from https://github.com/etcd-io/etcd/pull/10913.)

Propose v3.5 support for having a cluster automatically promote
a learner node to a voting member when added to a cluster with
`member add --learner --auto`.

The changes proposed for v3.5 in this PR are implemented in https://github.com/etcd-io/etcd/pull/10887

### CODE REVIEW

**Review by ptabor — APPROVED** [review]
Thank you. It looks good to me.

**Comment by nate-double-u** [comment]
@maxenglander, thanks for this. Could you resolve the conflict by adding this to the `content/docs/next` folder rather than `current`?

**Comment by maxenglander** [comment]
@nate-double-u sure, done!

**Comment by jberkus** [comment]
@ptabor @nate-double-u auto-promoting learners didn't merge.  

Leaving on hold while we address the original feature.

**Comment by jmhbnz** [comment]
Closing this stale pull request as part of triage.  We can always re-open in future if needed once the relevant feature is completed.  Noting that there are conflicts in this pr so it might be easiest to just start fresh once the feature is merged.

**Pull request closed without merging**
